app/models/user.rb

```diff
@@ -82,8 +85,13 @@ def self.reflect_on_association(association_name)
     end
   end
 
-  def self.find_for_database_authentication(conditions = {})
-    where('"users"."display_name" ILIKE ?', conditions[:display_name]).first || super
+  def self.find_for_database_authentication(conditions = { })
+    conditions = conditions.dup
```

Comment:
this just reads strangely to me, can you rename the method parameter to be warden_conditions? 

https://github.com/plataformatec/devise/wiki/How-To:-Allow-users-to-sign-in-using-their-username-or-email-address#for-activerecord